cmd/machine-api-operator/start.go

```diff
@@ -75,6 +81,16 @@ func runStartCmd(cmd *cobra.Command, args []string) {
 }
 
 func startControllers(ctx *ControllerContext) error {
```

Comment:
Can you create additional function `initRecorder`:
```go
func initRecorder(kubeClient kubernetes.Interface) eventBroadcaster.EventRecorder {
	eventRecorderScheme := runtime.NewScheme()
	osconfigv1.Install(eventRecorderScheme)
	eventBroadcaster := record.NewBroadcaster()
	eventBroadcaster.StartLogging(glog.Infof)
	eventBroadcaster.StartRecordingToSink(&coreclientsetv1.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
	return eventBroadcaster.NewRecorder(eventRecorderScheme, v1.EventSource{Component: "machineapioperator"})
}
```
, change `startControllers` implementation into:

```go
func startControllers(ctx *ControllerContext) {
	kubeClient := ctx.ClientBuilder.KubeClientOrDie(componentName)
	recorder := initRecorder(kubeClient)
	go operator.New(
		componentNamespace, componentName,
		startOpts.imagesFile,
		config,
		ctx.KubeNamespacedInformerFactory.Apps().V1().Deployments(),
		kubeClient,
		ctx.ClientBuilder.OpenshiftClientOrDie(componentName),
		recorder,
	).Run(2, ctx.Stop)
}
```

and replace:

```go
if err := startControllers(ctrlCtx); err != nil {
	glog.Fatalf("error starting controllers: %v", err)
}
```
with `startControllers(ctrlCtx)`?

`startControllers` always returns `nil` so it does not make sense to check for `err != nil` and fail.

Reply:
done

Reply:
`startControllers` did not change

Reply:
sorry forgot that I disabled auto-save option

Reply:
Thank you. Can you also remove `error` return type of the function? And remove `return nil`?

Reply:
done